services/spam-ban/spam-banning.service.js

```diff
@@ -0,0 +1,108 @@
+const { EmbedBuilder } = require("discord.js");
+const config = require("../../config");
+
+class SpamBanningService {
+  static async handleInteraction(interaction) {
+    const message = interaction.options.getMessage("message");
+    if (message.author.bot || SpamBanningService.#isAdmin(message.member)) {
+      interaction.reply({
+        content: "You do not have the permission to ban this user",
+        ephemeral: true,
+      });
+      return;
+    }
+
+    try {
+      let reply;
+
+      if (!message.member) {
+        message.react("❌");
```

Comment:
I'm assuming  we're reacting to the banned message? So this reaction will be done on a message in a public channel? Not sure we'd want that, but do confirm with others though.

Reply:
The button will be mainly used on automod messages that are blocked and logged, Which only staff can see. We wont be using this for messages that automod fails to block or doesnt delete. We will use manual modding to make sure that any left over spam msgs will be deleted by us.

Reply:
Okay, so this command is _only_ for the automod messages. Isn't automod a bot though? Won't `message.author.bot` always return true? How are we able to get the offending user's id from an automod message? I'm drawing blanks on the documentation, so some links could be helpful 😅 

I definitely should set up automod in a dummy server to test this myself instead of bombarding this PR lol

Reply:
Those are valid questions which I myself had, and made up an environment to explore. To my surprise however is that the Autobot logs actually act as if those message were posted by the user. message.author is actually the user who's message is blocked by Automod. Yeah certainly set up an environment and test things out